src/extension/app-schema/app-schema-web/src/main/java/org/geoserver/complex/ComplexToSimpleOutputHandler.java

```diff
@@ -0,0 +1,94 @@
+import org.opengis.feature.type.Name;
+import org.xml.sax.helpers.NamespaceSupport;
+
+/**
+ * Performs the complex to simple feature collection transformation returning the wrapped feature
+ * collection.
+ */
+public class ComplexToSimpleOutputHandler {
+
+    private static final Logger LOG = Logging.getLogger(ComplexToSimpleOutputHandler.class);
+
+    private final FeatureCollectionResponse result;
+    private final Catalog catalog;
+
+    private final QName layerName;
+
+    public ComplexToSimpleOutputHandler(
+            Request request, FeatureCollectionResponse result, Catalog catalog) {
+        this.result = requireNonNull(result);
+        this.catalog = requireNonNull(catalog);
+        this.layerName = requireNonNull(ComplexToSimpleOutputCommons.getLayerName(request));
+    }
+
+    /** Performs the transformation and returns the decorated Feature collection . */
+    public Object execute() {
+        LOG.fine(() -> "Executing handler for layer name: " + layerName);
+        // get the rules map
+        Map<String, String> rulesMap = getRulesMap();
+        // get the feature collection
+        @SuppressWarnings("unchecked")
+        ComplexToSimpleFeatureCollection collection =
+                new ComplexToSimpleFeatureCollection(
+                        rulesMap, result.getFeature().get(0), buildNamespaceSupport(catalog));
+        // return the correct type based on the input collection response type
+        if (result instanceof FeatureCollectionResponse.WFS20) {
+            return new ComplexToSimpleFeatureCollectionResponse20(result, collection);
+        } else if (result instanceof FeatureCollectionResponse.WFS11) {
+            return new ComplexToSimpleFeatureCollectionResponse11(result, collection);
+        }
+        return collection;
+    }
+
+    /** Returns the transformation rules map from the GeoServer layer configurations. */
+    @SuppressWarnings("unchecked")
+    private Map<String, String> getRulesMap() {
+        Name name = new NameImpl(layerName);
+        LayerInfo layerInfo = catalog.getLayerByName(name);
+        MetadataMap metadataMap = layerInfo.getMetadata();
+        Map<String, String> map =
+                metadataMap.get(ComplexToSimpleOutputCommons.RULES_METADATAMAP_KEY, Map.class);
+        if (map == null) {
+            LOG.fine(() -> "Rules map not found on layer: " + layerInfo);
+            return Collections.emptyMap();
+        }
+        LOG.fine(() -> "Rules map found: " + map);
+        return map;
+    }
+
+    /**
+     * Builds the {@link NamespaceSupport} instance from the catalog available namespaces.
+     *
+     * @param catalog the GeoServer catalog
+     * @return the resulting instance
+     */
+    private NamespaceSupport buildNamespaceSupport(Catalog catalog) {
+        List<NamespaceInfo> namespaces = catalog.getNamespaces();
+        NamespaceSupport nsSupport = new NamespaceSupport();
+        for (NamespaceInfo namespaceInfo : namespaces) {
```

Comment:
Just thinking out loud here... is this enough? Does one have a workspace for all secondary namespaces when setting up an app-schema mapping?